woocommerce/woocommerce-gutenberg-products-block
# add extra checks to checkout/cart/order APIs to avoid editor errors

Fixes #2071

This PR fixes an issue with checkout block in editor – various order-related state is not available in editor/admin.

- Add escape clause to `Checkout::get_route_response()` - bail if `is_admin()`. Also add some redundant paranoid checks on `WC()` `cart` & `session` within methods used in that API response.
- Adds some default state to cart-checkout context constants where needed. 

Also an unrelated fix, issue introduced when I rebased a previous PR (missed before merging).

- Removes a redundant `register_block_script()` call in `src/BlockTypes/Checkout.php` `render()`.

### How to test the changes in this Pull Request:

1. Test checkout block in editor.
2. Test checkout block in front end.

Should be able to add the block, edit settings, publish, reload editor, etc. And should function correctly on front end.
